src/main/java/de/mediathekview/mserver/crawler/srf/parser/SrfFilmJsonDeserializer.java

```diff
@@ -0,0 +1,351 @@
+            result.id = chapterListEntry.get(ATTRIBUTE_ID).getAsString();
+          }
+
+          if (chapterListEntry.has(ATTRIBUTE_DURATION)) {
+            long duration = chapterListEntry.get(ATTRIBUTE_DURATION).getAsLong();
+            result.duration = Duration.of(duration, ChronoUnit.MILLIS);
+          }
+
+          if (chapterListEntry.has(ATTRIBUTE_DESCRIPTION)) {
+            result.description = chapterListEntry.get(ATTRIBUTE_DESCRIPTION).getAsString();
+          }
+
+          if (chapterListEntry.has(ELEMENT_RESOURCE_LIST)) {
+            result.videoUrl = parseResourceList(chapterListEntry.get(ELEMENT_RESOURCE_LIST));
+          }
+          
+          if (chapterListEntry.has(ELEMENT_SUBTITLE_LIST)) {
+            result.subtitleUrl = parseSubtitleList(chapterListEntry.get(ELEMENT_SUBTITLE_LIST));
+          }
+        }
+      }
+    }
+
+    return result;
+  }
+  
+  private static String parseSubtitleList(JsonElement aSubtitleListElement) {
+    if (!aSubtitleListElement.isJsonArray()) {
+      return "";
+    }
+
+    JsonArray subtitleArray = aSubtitleListElement.getAsJsonArray();
+    for (JsonElement arrayItemElement : subtitleArray) {
+      if (!arrayItemElement.isJsonNull()) {
+        JsonObject arrayItemObject = arrayItemElement.getAsJsonObject();
+
+        if (arrayItemObject.has(ATTRIBUTE_FORMAT) && arrayItemObject.has(ATTRIBUTE_URL)) {
+          if (arrayItemObject.get(ATTRIBUTE_FORMAT).getAsString().equals(SUBTITLE_FORMAT)) {
+            return arrayItemObject.get(ATTRIBUTE_URL).getAsString();
+          }
+        }
+      }
+    }
+
+    return "";
+  }
+
+  private static String parseResourceList(JsonElement aResourceListElement) {
+    if (!aResourceListElement.isJsonArray()) {
+      return "";
+    }
+
+    JsonArray resourceArray = aResourceListElement.getAsJsonArray();
+    for (JsonElement arrayItemElement : resourceArray) {
+      if (!arrayItemElement.isJsonNull()) {
+        JsonObject arrayItemObject = arrayItemElement.getAsJsonObject();
+
+        if (arrayItemObject.has(ATTRIBUTE_MIMETYPE) && arrayItemObject.has(ATTRIBUTE_URL)) {
+          if (arrayItemObject.get(ATTRIBUTE_MIMETYPE).getAsString().contains("x-mpegURL")) {
+            return arrayItemObject.get(ATTRIBUTE_URL).getAsString();
+          }
+        }
+      }
+    }
+
+    return "";
+  }
+
+  private Map<Resolution, String> readUrls(String aM3U8Url) {
+    Map<Resolution, String> urls = new HashMap<>();
+
+    MVHttpClient mvhttpClient = MVHttpClient.getInstance();
+    OkHttpClient httpClient = mvhttpClient.getHttpClient();
+    Request request = new Request.Builder()
+            .url(aM3U8Url).build();
+    try (okhttp3.Response response = httpClient.newCall(request).execute()) {
+      if (response.isSuccessful()) {
+        String content = response.body().string();
+
+        M3U8Parser parser = new M3U8Parser();
+        List<M3U8Dto> m3u8Data = parser.parse(content);
+        m3u8Data.forEach((entry) -> {
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Remove the parentheses around the "entry" parameter (sonar.java.source not set. Assuming 8 or greater.) [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1611)